packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -182,24 +182,25 @@ export class Outbox {
 		}
 
 		const compressedBatch = this.params.compressor.compressBatch(batch);
-		if (compressedBatch.contentSizeInBytes <= this.params.config.maxBatchSizeInBytes) {
-			// If we don't reach the maximum supported size of a batch, it can safely be sent as is
-			return compressedBatch;
-		}
 
 		if (this.params.splitter.isBatchChunkingEnabled) {
-			return this.params.splitter.splitCompressedBatch(compressedBatch);
+			return compressedBatch.contentSizeInBytes <= this.params.splitter.chunkSizeInBytes
```

Comment:
Will this take into account op envelope size estimates when this and your other PR merge together?

Reply:
No. I mean not right here. Chunking will not help if the op envelope size overhead makes to batch too large. This value will be used (in the other PR) for deciding if chunking adds an extra op or not.